### PR TITLE
drains tests

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_multiply.cfg [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#####
+# API(s) being tested: [drains]value=
+##
+# Actions:
+# Give the leaders a drains ability with both the value and multiply attributes
+# Have the side 1 leader attack the side 2 leader with both its weapons
+##
+# Expected end state:
+# The both leaders have 2 weapons each of which now drains 8 hp (2*4)
+# The leader of side 1 heals 6 hp total since it has full hp when its first strike is made
```
Comment (ProditorMagnus) [inline data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_multiply.cfg:12]: either this or previous line is wrong

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_no_value.cfg [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        [modify_unit]
+            [filter]
+            [/filter]
+            [effect]
+                apply_to = new_ability
+                [abilities]
+                    [drains]
+                        id = "test-drains"
+                        name = _"test-drains"
+                        affect_self = yes
```
Comment (ProditorMagnus) [inline data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_no_value.cfg:28]: personally I like this style of only putting relevant attributes more than TEST_ABILITY

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_multiply.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_multiply.cfg:12]: Yeah, I copied and updated a previous set of tests and I missed updating that.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_no_value.cfg (hunk 2 above)
Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/drains/drains_no_value.cfg:28]: It's a bunch of duplicate code, which is what macros are made to help deal with. TEST_ABILITY also makes sure all attributes are set with the intended value and gives a standard ID for the ability.

**Pull request merged**